tools/block-generator/generator/generate_test.go

```diff
@@ -191,6 +191,49 @@ func TestAssetDestroy(t *testing.T) {
 	require.Len(t, g.assets, 0)
 }
 
+func TestAppCreate(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	g := makePrivateGenerator(t, 0, bookkeeping.Genesis{})
+
+	// app call transaction creating appBoxes
+	actual, txn, err := g.generateAppCallInternal(appBoxesCreate, 1, 0, 0, nil)
+	require.NoError(t, err)
+	require.Equal(t, appBoxesCreate, actual)
+	require.Equal(t, protocol.ApplicationCallTx, txn.Type)
+	require.Len(t, g.apps, 0)
+	require.Len(t, g.pendingApps, 1)
+	require.Len(t, g.pendingApps[appKindBoxes], 1)
+	require.Len(t, g.pendingApps[appKindSwap], 0)
+	require.Len(t, g.pendingApps[appKindBoxes][0].holdings, 1)
+	require.Len(t, g.pendingApps[appKindBoxes][0].holders, 1)
+	ad := *g.pendingApps[appKindBoxes][0]
+	holding := *ad.holdings[0]
+	require.Equal(t, holding, *ad.holders[0])
+	require.Equal(t, uint64(1001), holding.appIndex)
+	require.Equal(t, ad.appID, holding.appIndex)
+	require.Equal(t, appKindBoxes, ad.kind)
+
+	// app call transaction creating appSwap
+	actual, txn, err = g.generateAppCallInternal(appSwapCreate, 1, 0, 0, nil)
+	require.NoError(t, err)
+	require.Equal(t, appSwapCreate, actual)
+	require.Equal(t, protocol.ApplicationCallTx, txn.Type)
+	require.Len(t, g.apps, 0)
+	require.Len(t, g.pendingApps, 2)
+	require.Len(t, g.pendingApps[appKindBoxes], 1)
+	require.Len(t, g.pendingApps[appKindSwap], 1)
+	require.Len(t, g.pendingApps[appKindSwap][0].holdings, 1)
+	require.Len(t, g.pendingApps[appKindSwap][0].holders, 1)
+	ad = *g.pendingApps[appKindSwap][0]
+	holding = *ad.holdings[0]
+	require.Equal(t, holding, *ad.holders[0])
+	require.Equal(t, uint64(1001), holding.appIndex)
```

Comment:
I'm surprised the appID is still 1001. I expected to have incremented to 1002.

Reply:
This is because I'm not passing the new `intra` into `generateAppCallInternal()`. I'm much more explicit about in the next PR. I propose keeping this thread open and adding it as a **TODO** to the follow up.